### ISSUE TITLE
auto-run documented tool programs

### ISSUE THREAD

**Issue by markusicu** [unicode-org/unicodetools#187]
In a CI workflow, run each documented tool program, like we currently run org.unicode.text.UCD.Main. Each tool should run successfully (not crash, not fail). For this issue, ignore the output.
- Look through the https://github.com/unicode-org/unicodetools/tree/main/docs pages (excluding "help")
- For each tool documented there,
  - Add it to the list below
  - Add it to CI

List of tools:
- [x] AacOrder.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/emoji/aac.md#aacorderjava
- [ ] EmojiFrequency.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/emoji/emoji-frequency.md#processing
  - May need a dummy .tsv file
- [x] org.unicode.text.UCA.Main, see https://github.com/unicode-org/unicodetools/blob/main/docs/uca/index.md#tools--tests
- [x] GenerateIdna.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/idna.md
- [x] GenerateIdnaTest.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/idna.md
- [x] GenerateEnums.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/newunicodeproperties.md#run-generateenumsjava
  - Would be useful to fail if the generated output differs from the repo files.
- [x] GenerateConfusables, see https://github.com/unicode-org/unicodetools/blob/main/docs/security.md#generating
- [x] GenerateUnihanCollators, see https://github.com/unicode-org/unicodetools/blob/main/docs/unihan.md#run-generateunihancollators
- [x] CheckProperties, see https://github.com/unicode-org/unicodetools/blob/main/docs/newunicodeproperties.md#checking-other-properties

Not a tool:
- ToolUnicodePropertySource.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/changing-ucd-properties.md#toolunicodepropertysourcejava

Skip:
- GenerateUnihanCollatorFiles, see https://github.com/unicode-org/unicodetools/blob/main/docs/unihan.md#run-generateunihancollatorfiles
- GenerateEmoji.java, see https://github.com/unicode-org/unicodetools/blob/main/docs/emoji/generate-emoji.md#running-the-tool
  - 2022-mar-22: out of scope because it needs access to another repo, see [discussion below](https://github.com/unicode-org/unicodetools/issues/187#issuecomment-1075566958)

**Comment by srl295** [comment]
depends on knowing what the tools are => https://github.com/unicode-org/unicodetools/issues/191

**Comment by markusicu** [comment]
Josh to find tools, Elango to help add them to CI.

**Comment by markusicu** [comment]
Example: https://github.com/unicode-org/unicodetools/blob/main/docs/idna.md
--> add an entry in the issue description/checklist above for GenerateIdna.java
--> also add an entry for GenerateIdnaTest.java

**Comment by markusicu** [comment]
Hi, just checking -- any progress? I think "Josh to find tools" comes first...?

**Comment by josh-hadley** [comment]
@markusicu @srl295 I went through and noted what appeared to be Tool-like items and added to the checklist above. Some of them might not be so easy (or appropriate) to auto-run.

**Comment by markusicu** [comment]
Thanks @josh-hadley ! Taking a look...
- EmojiFrequency: We don't seem to have its input file vendorRaw.tsv in the repo. Provide a dummy file?
- [ToolUnicodePropertySource](https://github.com/unicode-org/unicodetools/blob/main/unicodetools%2Fsrc%2Fmain%2Fjava%2Forg%2Funicode%2Ftext%2FUCD%2FToolUnicodePropertySource.java): This is not a tool. No `main()`, and it's a class that provides properties to the tools, rather than being a tool.
- [GenerateEnums](https://github.com/unicode-org/unicodetools/blob/main/unicodetools%2Fsrc%2Fmain%2Fjava%2Forg%2Funicode%2Fprops%2FGenerateEnums.java): This tool wants to overwrite two .java files in the repo source. For CI, we could add an option to have it write to /tmp or similar.
- [GenerateUnihanCollatorFiles](https://github.com/unicode-org/unicodetools/blob/main/unicodetools%2Fsrc%2Fmain%2Fjava%2Forg%2Funicode%2Fdraft%2FGenerateUnihanCollatorFiles.java): This is a tiny tool that inserts the output data from GenerateUnihanCollators into CLDR data files. Probably ok not to add to CI.

@echeran do you feel unblocked?

**Comment by echeran** [comment]
@markusicu yes, that should unblock me

**Comment by markusicu** [comment]
- EmojiFrequency.java -- using non-public files from vendors, use stub files for smoke testing
- GenerateEmoji.java needs a path to a separate, private "images" repo -- maybe use a workflow that is only invoked manually; could use a "deploy key" (GitHub secrets) on the images repo to ssh-fetch the images repo
- UCA.Main -- run on data/dev files

**Comment by markusicu** [comment]
@echeran EmojiFrequency.java still TBD?

**Comment by echeran** [comment]
> @echeran EmojiFrequency.java still TBD?

@markusicu Yep, still need a dummy .tsv file from @macchiati for running that in CI.

**Comment by markusicu** [comment]
> > > @echeran EmojiFrequency.java still TBD?
> > 
> > @markusicu Yep, still need a dummy .tsv file from @macchiati for running that in CI.
> 
> @macchiati did you provide a dummy .tsv file to @echeran for running this tool?

ping @macchiati

**Comment by macchiati** [comment]
I looked at this, and it would take some work. So I would put it at a low
priority.

**Comment by markusicu** [comment]
> I looked at this, and it would take some work. So I would put it at a low priority.

Creating a dummy .tsv file would take some work? How much? An hour? An afternoon? A couple of days?
The goal is to routinely run EmojiFrequency.java so that we catch if & when it gets broken, not next time we *need* to run it.

**Comment by markusicu** [comment]
Closing this issue as done. The one remaining tool (EmojiFrequency.java) will be tracked via new issue #338.

**Comment by macchiati** [comment]
I worked on it a bit, but it takes more than just a dummy file.
